external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -564,20 +575,38 @@ private[kafka010] object KafkaSourceProvider extends Logging {
 
   def getKafkaOffsetRangeLimit(
       params: CaseInsensitiveMap[String],
+      globalOffsetTimestampOptionKey: String,
       offsetByTimestampOptionKey: String,
       offsetOptionKey: String,
       defaultOffsets: KafkaOffsetRangeLimit): KafkaOffsetRangeLimit = {
-    params.get(offsetByTimestampOptionKey).map(_.trim) match {
-      case Some(json) => SpecificTimestampRangeLimit(JsonUtils.partitionTimestamps(json))
-      case None =>
-        params.get(offsetOptionKey).map(_.trim) match {
-          case Some(offset) if offset.toLowerCase(Locale.ROOT) == "latest" =>
-            LatestOffsetRangeLimit
-          case Some(offset) if offset.toLowerCase(Locale.ROOT) == "earliest" =>
-            EarliestOffsetRangeLimit
-          case Some(json) => SpecificOffsetRangeLimit(JsonUtils.partitionOffsets(json))
-          case None => defaultOffsets
-        }
+    // The order below represents "preferences"
```

Comment:
`specifying both global timestamp and specific timestamp for partition` added to test case 1. vs 2. fallback which is good. In order to cover all "preferences" maybe we can add a case where all config options added.

Reply:
Yeah I changed the test a bit to cover two cases: all options / timestamp per partition vs offset. Thanks for the suggestion.